docs/blueprints/EP036.rst

```diff
@@ -0,0 +1,164 @@
+    | (wildcard match) |         |                  |         |                  |         |                  |         |                  |         |                  |
+    |      of_lldp     |  ---->  |                  |  ---->  |   (exact match   |  ---->  |    (exact match  |  ---->  |      (exact      |  ---->  |    (wildcard)    |
+    |      coloring    |  ---->  |                  |  ---->  | in_port + vlan)  |  ---->  |      in_port)    |  ---->  |       match)     |  ---->  |                  |
+    |                  |         |                  |         |                  |         |                  |         |                  |         |                  |
+    |                  |         |                  |         |                  |         |                  |         |                  |         |                  |
+    |   goto_table: 1  |         |   goto_table: 2  |         |   goto_table: 3  |         |   goto_table: 5  |         |                  |         |                  |
+    +------------------+         +------------------+         +------------------+         +------------------+         +------------------+         +------------------+
+
+New collection, **tables**
+==========================
+
+Initially proposed here is a datastructured representing a linked list (without cycles)
+
+  .. code:: javascript
+
+    "pipeline": {
+      0: {
+        "table_miss_flow": {
+          "priority": 123, "match": {},
+          "instructions": [{"instruction_type": "goto_table", "table_id": 1}]
+        },
+        "napps_table_groups": {"coloring": ["base"],"of_lldp": ["base"]}
+      },
+      1: {
+        "table_miss_flow": {
+          "priority": 123, "match": {},
+          "instructions": [{"instruction_type": "goto_table","table_id": 2}]
+        },
+        "napps_table_groups": {"flow_manager": []}
+      },
+      2: {
+        "table_miss_flow": {
+          "priority": 123, "match": {},
+          "instructions": [{"instruction_type": "goto_table","table_id": 3}]
+        },
+        "napps_table_groups": {"mef_eline": ["evpl"]}
+      },
+      3: {
+        "table_miss_flow": {
+          "priority": 123, "match": {},
+          "instructions": [{"instruction_type": "goto_table","table_id": 5}]
+        },
+        "napps_table_groups": {"mef_eline": ["epl"]}
+      }
+      4: {
+        "table_miss_flow": {},
+        "napps_table_groups": {"INT": []}
+      }
+      5: {
+        "table_miss_flow": {},
+        "napps_table_groups": {}
+      }
+    }
+
+Events
+======
+
+Subscribed
+-----------
+  - **kytos/flow_manager.flow.added**
+  - **kytos/of_core.handshake.completed**
+
+Published
+---------
+
+kytos/multi_table.deploy_table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event should sent before any NApp sends its first flow to be published.

```

Comment:
Based on the way you're proposing the changes, it ended up very well, in fact, it has the potential to change at any time during runtime since you've figured out a nicer way to install and delete the flows.

Reply:
So, could we introduce and officialize a new endpoint to allow a pipeline to be set (and activated)?

Reply:
Yeah, it will need get, delete and post (create, edit, deploy). Just one thing, I have been using the name `deploy` but the user have been using `enable` and `disable` for other requests. Is it okay to use it here too when changing pipeline set ups? wdyt
Before I thought that the user will only deploy a table but the opposite could mean to get rid of the tables and set everything to table 0.

Reply:
Yes, on `topology`'s and `of_lldp`'s endpoints we ended up using `/enable` and `/disable` on routes, following that pattern would be great. Let's also specify all the states that a pipeline can have though.

> Before I thought that the user will only deploy a table but the opposite could mean to get rid of the tables and set everything to table 0.

Indeed. Yes, let's always operate on an entire pipeline level, but then when computing what needs to be installed/deleted you can compare if table groups stayed the same or changed (moved, added or removed). Comparing what changed is an optimization that's highly desired to have since it'll minimize IO.